### ISSUE TITLE
Adicionar sessão de ferramentas de build e/ou gerenciamento de pacotes

### ISSUE THREAD

**Issue by anikarni** [ThoughtWorksInc/guia-de-desenvolvimento-tecnico#57]
No momento, em sessão alguma comentamos sobre ferramentas como gradle, npm, gems, etc. 

Precisamos ou incluir dentro de alguma sessão ou criar sua própria sessão. O que acham?

**Comment by schmidt-marcelo** [comment]
Meus R$0,02:
- Se incluirmos essas ferramentas em alguma sessão (Entrega Contínua) faz sentido uma vez que são esses processos são construídos em cima desse "conceito", porém não é exclusividade desses processos, então pode ficar confuso, difícil de encontrar;
- Se criarmos uma sessão para isso, fica um pouco dissociado do processo de desenvolvimento de software, mas fica mais fácil de achar e mais concentrado.

Eu particularmente acho que uma sessão separada para essas ferramentas seja o mais indicado por enquanto, mas precisamos lembrar que elas sozinhas não tem motivos de existir. 

Faz sentido?

**Comment by anikarni** [comment]
Pessoal, 

Tava reconsiderando esse issue e continuo na dúvida. Devemos mesmo ter uma sessão que se preocupa só com ferramentas? Não deveria fazer parte de um de dois conceitos: Entrega de Software (talvez na subsessao "Entregue seu código continuamente") ou Infraestrutura (adicionando a "infra com código")? Talvez minha ignorância nesse tópico esteja me impedindo de tomar uma decisão lógica.

**Comment by schmidt-marcelo** [comment]
"infra com código" tem mais a ver com DevOps. 

Build tools, como eu falei em outro comment, não estão necessariamente atrelados a DevOps, apesar deste último fazer uso muitas vezes dessas ferramentas. 

Acho que  tem mais a ver com Entrega de Software.

**Comment by anikarni** [comment]
Concordo, mas então eliminamos a possibilidade de ter eles como uma sessão separada?

**Comment by schmidt-marcelo** [comment]
Depende, @anikarni . Na minha visão não existe entrega de software sem ferramentas de construção, entrega e ferramentas de construção formam uma composição. 

Claro que poderíamos fazer tudo "na mão" para entrega, mas o custo de se fazer isso é muito mais alto do que assumir que entrega depende de ferramentas de construção. 

Sendo assim, na minha visão meio míope talvez, eu acho que faz parte da Entrega/Integração de Software.

**Comment by schmidt-marcelo** [comment]
Pois é, complicado decidir.

Em uma resposta anterior, comentei que poderia ser uma seção separada. Porém pensando melhor, ferramentas de construção casam bem com Entrega de Software. 

Na minha visão não existe entrega de software sem ferramentas de construção, entrega e ferramentas de construção formam uma composição.

Claro que poderíamos fazer tudo "na mão" para entrega, mas o custo de se fazer isso é muito mais alto do que assumir que entrega depende de ferramentas de construção.

Sendo assim, na minha visão meio míope talvez, voltando atrás em relação ao meu primeiro comentário, eu acho que faz parte da Entrega/Integração de Software.

**Comment by MarcosX** [comment]
E se antes na sessão Entrega Contínua, antes de `Integre seu código continuamente` adicionarmos uma sessão `Garanta que o ambiente é fácil de reproduzir`, ou algo do tipo, e lá colocamos os conceitos de gerenciamento de dependências (destacando que cada linguagem possui suas próprias ferramentas) e ainda podia entrar um pouco em provisionamento/virtualização (talvez falar sobre usar Docker para desenvolvimento)?

**Comment by anikarni** [comment]
👍 pra 'Entrega Contínua'
